test/helm_basic_example_integration_test.go

```diff
@@ -0,0 +1,97 @@
+	t.Parallel()
+
+	// Path to the helm chart we will test
+	helmChartPath, err := filepath.Abs("../examples/helm-basic-example")
+	require.NoError(t, err)
+
+	// Setup the kubectl config and context. Here we choose to use the defaults, which is:
+	// - HOME/.kube/config for the kubectl config file
+	// - Current context of the kubectl config file
+	kubectlOptions := k8s.NewKubectlOptions("", "")
+
+	// To ensure we can reuse the resource config on the same cluster to test different scenarios, we setup a unique
+	// namespace for the resources for this test.
+	// Note that namespaces must be lowercase.
+	namespaceName := fmt.Sprintf("helm-basic-example-%s", strings.ToLower(random.UniqueId()))
+	k8s.CreateNamespace(t, kubectlOptions, namespaceName)
+	// Make sure we set the namespace on the options
+	kubectlOptions.Namespace = namespaceName
+	// ... and make sure to delete the namespace at the end of the test
+	defer k8s.DeleteNamespace(t, kubectlOptions, namespaceName)
+
+	// Setup the args. For this test, we will set the following input values:
+	// - containerImageRepo=nginx
+	// - containerImageTag=1.15.8
+	options := &helm.Options{
+		KubectlOptions: kubectlOptions,
+		SetValues: map[string]string{
+			"containerImageRepo": "nginx",
+			"containerImageTag":  "1.15.8",
+		},
+	}
+
+	// We generate a unique release name so that we can refer to after deployment.
+	// By doing so, we can schedule the delete call here so that at the end of the test, we run
+	// `helm delete RELEASE_NAME` to clean up any resources that were created.
+	releaseName := fmt.Sprintf(
+		"nginx-service-%s",
+		strings.ToLower(random.UniqueId()),
+	)
+	defer helm.Delete(t, options, releaseName, true)
+
+	// Deploy the chart using `helm install`. Note that we use the version without `E`, since we want to assert the
+	// install succeeds without any errors.
+	helm.Install(t, options, helmChartPath, releaseName)
+
+	// Now let's verify the deployment. We will get the service endpoint and try to access it.
+
+	// First we need to get the service name. We will use domain knowledge of the chart here, where the name is
+	// RELEASE_NAME-CHART_NAME
+	serviceName := fmt.Sprintf("%s-helm-basic-example", releaseName)
+
+	// Next we wait until the service is available. This will wait up to 10 seconds for the service to become available,
+	// to ensure that we can access it.
+	k8s.WaitUntilServiceAvailable(t, kubectlOptions, serviceName, 10, 1*time.Second)
+
+	// Now we verify that the service will successfully boot and start serving requests
+	service := k8s.GetService(t, kubectlOptions, serviceName)
+	endpoint := k8s.GetServiceEndpoint(t, service, 80)
+	// Test the endpoint for up to 5 minutes. This will only fail if we timeout waiting for the service to return a 200
+	// response.
+	http_helper.HttpGetWithRetryWithCustomValidation(
+		t,
+		fmt.Sprintf("http://%s", endpoint),
+		30,
+		10*time.Second,
+		func(statusCode int, body string) bool {
+			return statusCode == 200
```

Comment:
Check the body contains the text "nginx"?

Reply:
Ok